### PR TITLE
ENH Take advantage of auto-scaffolded form fields

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Issue
- https://github.com/silverstripe/.github/issues/262

### CODE REVIEW

File: src/Model/Subsite.php [hunk 1]
```diff
@@ -666,16 +667,10 @@ public function canEdit($member = false)
...
-                        '',
-                        $this->Domains(),
-                        GridFieldConfig_RecordEditor::create(10)
-                    )
-                ]);
+            $domainsGridField = $fields->dataFieldByName('Domains');
+            if ($domainsGridField instanceof GridField) {
+                $domainsGridField->setTitle('');
+                $domainsGridField->getConfig()->removeComponentsByType(GridFieldAddExistingAutocompleter::class);
             }
```
Comment (ChloeSartorelli) [inline src/Model/Subsite.php:674]: Uses the existing gridfield instead of building it from scratch.

**Comment by GuySartorelli** [comment]
JS CI failure is unrelated to these changes

File: src/Model/Subsite.php [hunk 2]
```diff
@@ -666,16 +667,10 @@ public function canEdit($member = false)
...
-                        'Domains',
-                        '',
-                        $this->Domains(),
-                        GridFieldConfig_RecordEditor::create(10)
-                    )
-                ]);
+            $domainsGridField = $fields->dataFieldByName('Domains');
+            if ($domainsGridField instanceof GridField) {
+                $domainsGridField->setTitle('');
+                $domainsGridField->getConfig()->removeComponentsByType(GridFieldAddExistingAutocompleter::class);
```
Comment (emteknetnz) [inline src/Model/Subsite.php:673]: Why are we removing the GridFieldAddExistingAutocompleter?

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: src/Model/Subsite.php (hunk 2 above)
Comment (ChloeSartorelli) [reply src/Model/Subsite.php:673]: To match the gridfield component set that existing prior to this PR.

**Review by emteknetnz — APPROVED** [review]
Tested locally, works well

Failing JS test is unrelated

**Pull request merged**